SERVER/src/narrator.py

```
@@ -1,14 +1,163 @@
+        """
+        if not self.stt_model:
+            print("Narrator: STT model not loaded. Cannot process narration.")
+            return {"text": "", "audio_path": audio_filepath, "speaker": self.default_speaker_name}
+
+        # Save a copy of the audio file to NARRATOR_AUDIO_PATH
+        try:
+            os.makedirs(NARRATOR_AUDIO_PATH, exist_ok=True)
+            ext = os.path.splitext(audio_filepath)[1]
+            unique_name = f"narration_{uuid.uuid4().hex}{ext}"
+            dest_path = os.path.join(NARRATOR_AUDIO_PATH, unique_name)
+            with open(audio_filepath, "rb") as src, open(dest_path, "wb") as dst:
```

Ruff (0.11.9)

Async functions should not open files with blocking methods like `open`

(ASYNC230)